### PR TITLE
Fix handling of safety switch across HALs

### PR DESCRIPTION
**Pull request by lucasdemarchi**
This removes some warnings/errors while initializing ArduPilot, but applies to any PX4-based boards.

### CODE REVIEW

**Comment by lucasdemarchi** [comment]
@khancyr where? I supposedly added where it was missing

**Comment by khancyr** [comment]
my bad, it is correct, it hadn't check that HAL_HAVE_SAFETY_SWITCH was already define for some board...

File: libraries/AP_BoardConfig/px4_drivers.cpp [hunk 1]
```diff
@@ -87,17 +87,17 @@ void AP_BoardConfig::px4_setup_pwm()
             AP_Param::set_default_by_name("RELAY_PIN", -1);
             AP_Param::set_default_by_name("RELAY_PIN2", -1);
         }
-#endif
 #endif
     }
+#endif
```
Comment (OXINARF) [inline libraries/AP_BoardConfig/px4_drivers.cpp]: Can we put this fix in the correct commit?

**Comment by lucasdemarchi** [comment]
> it's also missing FMUv4 macro define and VRBrain.

libraries/AP_HAL/board/vrbrain.h has the define for vrbrain already.

fmuv4 has SAFETY_IN which outrules  this define due to:
```c++
#if HAL_HAVE_SAFETY_SWITCH || defined(HAL_GPIO_PIN_SAFETY_IN)
```

**Comment by lucasdemarchi** [comment]
I fixed the rest. Apparently there's still a problem with sitl that declares to have safety switch but never handles it.

**Comment by OXINARF** [comment]
> libraries/AP_HAL/board/vrbrain.h has the define for vrbrain already.
>
>fmuv4 has SAFETY_IN which outrules this define due to:

@lucasdemarchi I think all the mixed use is very confusing, there is HAL_HAVE_SAFETY_SWITCH, HAL_GPIO_PIN_SAFETY_IN and AP_FEATURE_SAFETY_BUTTON. And there are also two other BOARD_SAFETY_* macros.

Seems like HAL_GPIO_PIN_SAFETY_IN is appropriate just for use inside the HAL and HAL_HAVE_SAFETY_SWITCH should be used outside and we wouldn't need an AP_FEATURE_SAFETY_BUTTON.

**Comment by lucasdemarchi** [comment]
I agree it's confusing. It was not me who did that way though. I **think** the idea is that if you declare a pin for safety switch, you sure have it, does it gets auto enabled.

As for AP_FEATURE_SAFETY_BUTTON, I agree. But from what I see @tridge introduced it probably to allow different builds for the same board (thus same hal). I think we can get rid of it and just defining the HAL variant accordingly, but @tridge should confirm.

**Comment by lucasdemarchi** [comment]
@OXINARF I pushed the one commit that was unrelated to safety switch, applied  what you asked and renamed the PR accordingly.

**Pull request merged**